pkgs/development/libraries/volk/default.nix

```
@@ -5,12 +5,19 @@
 , python3
 , enableModTool ? true
 , removeReferencesTo
+, fetchpatch
 }:
 
 stdenv.mkDerivation rec {
   pname = "volk";
   version = "2.5.0";
 
+  patches = lib.optional (stdenv.isDarwin && stdenv.isAarch64) [(fetchpatch {
+    url = "https://raw.githubusercontent.com/macports/macports-ports/e83a55ef196d4283be438c052295b2fc44f3df5b/science/volk/files/patch-cpu_features-add-support-for-ARM64.diff";
+    sha256 = "sha256-MNUntVvKZC4zuQsxGQCItaUaaQ1d31re2qjyPFbySmI=";
+    extraPrefix = "";
+  })];
```

```suggestion
  patches = [
    (fetchpatch {
      url = "https://raw.githubusercontent.com/macports/macports-ports/e83a55ef196d4283be438c052295b2fc44f3df5b/science/volk/files/patch-cpu_features-add-support-for-ARM64.diff";
      sha256 = "sha256-MNUntVvKZC4zuQsxGQCItaUaaQ1d31re2qjyPFbySmI=";
      extraPrefix = "";
    })
  ];
```
We want to prevent bit rot and patches goes after src.

---

Is there a reason that we want this patch to apply to all systems, not just `aarch64-darwin`?

---

Agreed.  Fixed.